src/screens/WasteCollectionScreen.js

```diff
@@ -143,6 +132,79 @@ export const WasteCollectionScreen = ({ navigation }) => {
     }
   );
 
+  const wasteAddressesStreetCount = globalSettings?.settings?.wasteAddresses?.streetCount || 5;
+  const wasteAddressesTwoStep = !!globalSettings?.settings?.wasteAddresses?.twoStep;
+
+  const getStreetString = useCallback(
+    (address) => {
+      if (wasteAddressesTwoStep) {
+        return address.street || '';
+      }
+
+      return `${address.street} (${address.zip} ${address.city})`;
+    },
+    [wasteAddressesTwoStep]
+  );
+
+  // show cities that contain the string in it
+  // return empty list on an exact match (except for capitalization)
+  const filterCities = useCallback(
+    (currentInputValue, addressesData) => {
+      if (currentInputValue === '') return [];
+
+      const cities = addressesData?.filter(
+        (address) =>
+          address.city !== currentInputValue &&
+          address.city.toLowerCase().includes(currentInputValue.toLowerCase())
+      );
```

Comment:
i think there should be a check for the `null` value of `city`.
like this:
`address.city?.toLowerCase()`

![IMG_8C3B5F24396B-1](https://user-images.githubusercontent.com/11755668/222449353-406d656e-3fd4-49fd-935e-7e20d45b4618.jpeg)